source/server/admin/stats_request.h

```diff
@@ -87,31 +108,42 @@ class StatsRequest : public Admin::Request {
   // current phase.
   void populateStatsForCurrentPhase(const ScopeVec& scope_vec);
 
-  // Populates all the metrics of the templatized type from scope_vec. Here we
-  // exploit that Scope::iterate is a generic templatized function to avoid code
-  // duplication.
-  template <class StatType> void populateStatsFromScopes(const ScopeVec& scope_vec);
+  virtual Stats::IterateFn<Stats::TextReadout> saveMatchingStatForTextReadout() PURE;
```

Comment:
I think these can be in the protected section.

Reply:
I revisited all of the access modifiers in `stats_request.h`, I realise I had unnecessarily "relaxed" some of the pre-existing access.